### DISCUSSION TITLE
How to set up material.yaml for hcp materials

### DISCUSSION THREAD

**Discussion by hugy888** [damask-multiphysics/DAMASK#16] [Q&A] [answered]
Hi, I have following setup for phase and homogenization in material.yaml for Aluminum:

`phase:
  Aluminum:
    lattice: cF
    mechanical:
      output: [F, P, F_p, O]
      elastic: {type: Hooke, C_11: 106750000000.0, C_12: 60410000000.0, C_44: 28340000000.0}

homogenization:
  direct:
    N_constituents: 1
    mechanical: {type: pass}
`

I was just wondering if we would like to run DAMASK simulation for hcp materials like Mg, would the setup be like this? We are only curious about Young's modulus and we remove the plastic part from mechanical.

`phase:
  Mg:
    lattice: hP
    mechanical:
      output: [F, P, F_p, O]
      elastic: {type: Hooke, C_11: 59.5e+9, C_12: 25.6e+9, C_44: 16.5e+9, C_33: 61.7e+9, C_13: 21.5e+9}

homogenization:
  direct:
    N_constituents: 1
    mechanical: {type: pass}
`

I added C_33 and C_13 for the elastic and change lattice into hP. Do we also need to add c/a ratio for the hcp material?

Any guidance on this will be greatly appreciated. Thanks!
Best regards

**Comment by hyharry** [comment]
you need to set `c/a` for your Mg phase for completeness (if not set, you will probably get an error when run), although elasticity problem solely does not use this value. you can try it out yourself.

**Reply by MarDiehl** [reply]
There are also many examples available on https://damask.mpie.de/documentation/files/config.tar.xz

**Reply by hugy888** [reply]
Hi @MarDiehl and @hyharry,
Thank you very much for your helpful response. I added the c/a ratio, and the simulation worked perfectly.
```
phase:
Mg:
lattice: hP
c/a: 1.623
mechanical:
output: [F, P, F_p, O]
elastic: {type: Hooke, C_11: 59.5e+9, C_12: 25.6e+9, C_44: 16.5e+9, C_33: 61.7e+9, C_13: 21.5e+9}

homogenization:
direct:
N_constituents: 1
mechanical: {type: pass}
```

From the examples you mentioned, I found the following elasticity information about Mg:
```
C_11: 59.5e+9
C_11,T: -1.930e+7
C_11,T^2: -1.215e+4

C_12: 25.6e+9
C_12,T: -2.216e+6
C_12,T^2: -4.138e+3

C_44: 16.5e+9
C_44,T: -1.006e+7
C_44,T^2: -7.692e+3

C_33: 61.7e+9
C_33,T: -2.175e+7
C_33,T^2: -5.624e+3

C_13: 21.5e+9
C_13,T: -1.921e+6
C_13,T^2: -4.283e+3
```

However, from other literature, I found that Mg could also have a unique C_66 tensor, with its elasticity tensor looking like this:
```
M = [[59.4e+9   25.61e+9  21.44e+9   0     0     0];...
        [ 25.61e+9  59.4e+9  21.44e+9   0     0     0];...
        [ 21.44e+9  21.44e+9  61.9e+9   0     0     0];...
        [   0    0   0  16.4e+9     0     0];...
        [   0    0   0   0    16.4e+9     0];...
        [   0    0   0   0     0    16.9e+9]];
```
I was wondering if for the material.yaml file, I should add the C_66 tensor like this:
```
phase:
Mg:
lattice: hP
c/a: 1.623
mechanical:
output: [F, P, F_p, O]
elastic: {type: Hooke, C_11: 59.5e+9, C_12: 25.61e+9, C_44: 16.4e+9, C_33: 61.9e+9, C_13: 21.44e+9, C_66: 16.9e+9}
```
It would be wonderful to have an example material.yaml file for hcp structures as a reference. Currently, most examples I could find are for fcc materials. Any guidance would be greatly appreciated!

Thank you again for your support!
Best regards

**Reply by MarDiehl** [reply]
hex has 5 independent stiffness constants, if you find literature data with C_66 it either just given for convenience or another constant (probably C_12) is not given. See https://www.sciencedirect.com/science/article/pii/S0010465521003921?via%3Dihub#br0100

We only have the current data. I believe for hex we have parameters for Ti and Mg. Getting elasticity parameters is easy, if you have (trustworthy) sources we can add them to the examples. Temperature-dependent parameters are already rather rare, and fitting crystal plasticity parameters requires a significant time investment.

**Reply by hugy888** [reply]
Hi @MarDiehl,
Thank you so much for your reply.

From the examples you mentioned, I found the following elasticity information about Mg:

```
type: Hooke

references:
  - L.J. Slutsky and C.W. Garland,
    Physical Review 107(4):972-976, 1957,
    https://doi.org/10.1103/PhysRev.107.972,
    fit to Tab. I (T_min=100K, T_max=300K)

C_11: 59.5e+9
C_11,T: -1.930e+7
C_11,T^2: -1.215e+4

C_12: 25.6e+9
C_12,T: -2.216e+6
C_12,T^2: -4.138e+3

C_44: 16.5e+9
C_44,T: -1.006e+7
C_44,T^2: -7.692e+3

C_33: 61.7e+9
C_33,T: -2.175e+7
C_33,T^2: -5.624e+3

C_13: 21.5e+9
C_13,T: -1.921e+6
C_13,T^2: -4.283e+3

T_ref: 293.15
```

I checked the literature by L.J. Slutsky and found a table of elastic constants of Mg at different temperatures, as shown below:
![image](https://github.com/user-attachments/assets/d2adaedc-e9f6-4247-aa77-3277745e553c)

It lists six different values. I was wondering if DAMASK treated C_66 the same as C_44 because their values were quite similar. Another issue is that I could not find the Cij values from the DAMASK Mg example in that table. As you mentioned, DAMASK fits curves for Mg, and predicts Cij values at 293.15 K?

Thanks again for your help!

**Reply by MarDiehl** [reply, marked as answer]
Instead of asking, just read the references (L.J. Slutsky and C.W. Garland and the one I shared). It's all explained there.
